src/main/java/cloud/fogbow/ras/core/ApplicationFacade.java

```diff
@@ -127,6 +127,9 @@ public String getPublicKey() throws InternalServerErrorException {
         try {
             SystemUser requester = authenticate(userToken);
             RasOperation rasOperation = new RasOperation(Operation.GET, ResourceType.CLOUD_NAME);
+            rasOperation.setTargetProvider(providerId);
+            rasOperation.setRequestingProvider(this.providerId);
```

Comment:
Perhaps we could change the constructor of RasOperation to get the requester and the provider as parameters, and obviate the need to set them afterwards?

Reply:
Currently RasOperation has 3 constructors. Should we add requester and provider parameters to all of them?

Reply:
Done.